tuf/download.py

```diff
@@ -192,33 +192,33 @@ def _download_file(url, required_length, fetcher, STRICT_REQUIRED_LENGTH=True):
   # This is the temporary file that we will return to contain the contents of
   # the downloaded file.
   temp_file = tempfile.TemporaryFile()
-  start_time = timeit.default_timer()
 
   average_download_speed = 0
   number_of_bytes_received = 0
 
   try:
-
-    for chunk in fetcher.fetch(url, required_length):
+    chunks = fetcher.fetch(url, required_length)
+    start_time = timeit.default_timer()
+    for chunk in chunks:
 
       stop_time = timeit.default_timer()
```

Comment:
I am wondering what exactly will happen between the `start_time` and `stop_time` given that they are just one after another.
Do `chunks` remain blocked until it's fully downloaded and then start the for loop (but that doesn't make sense, this would mean you have to have the whole file in memory which is what we want to escape)? 

Sorry for those probably obvious questions in this stage of the review process, but I wanted to make sure I understand this.

Reply:
good question. the blocking points should be:
* `fetcher.fetch()` -- connect(), TLS negotiation, etc
* each iteration of `chunks`: the iterator blocks until it has downloaded data for the chunk

this is the intention and "obvious implementation" at least: of course the Fetcher implementation could do things differently.

Reply:
> fetcher.fetch() -- connect(), DNS lookup, TLS negotiation, etc
This means that the `start_time` won't catch the delay when starting the fetch process which makes sense

> each iteration of chunks: the iterator blocks until it has downloaded data for the chunk

This means it will count the average download on each chunk which seems right.
Thanks, I think I understand it now!